policy/modules/apps/cdrecord.te

```diff
@@ -113,3 +132,98 @@ tunable_policy(`use_nfs_home_dirs',`
 optional_policy(`
 	resmgr_stream_connect(cdrecord_t)
 ')
+
+########################################
+#
+# Cdripper (e.g. cdda2wav) local policy
+#
+
+allow cdripper_t self:capability { ipc_lock sys_nice setgid setuid dac_override sys_rawio };
+allow cdripper_t self:process { getcap getsched setcap setrlimit setsched sigkill };
+allow cdripper_t self:fifo_file rw_fifo_file_perms;
+allow cdripper_t self:tcp_socket create_socket_perms;
+allow cdripper_t self:udp_socket create_socket_perms;
+allow cdripper_t cdripper_exec_t:file execute_no_trans;
+
+corecmd_exec_bin(cdripper_t)
+corecmd_exec_shell(cdripper_t)
+
+corenet_tcp_connect_all_unreserved_ports(cdripper_t)
+
+dev_list_all_dev_nodes(cdripper_t)
+dev_read_sysfs(cdripper_t)
+
+domain_interactive_fd(cdripper_t)
+domain_use_interactive_fds(cdripper_t)
+
+files_read_etc_files(cdripper_t)
+
+term_use_controlling_term(cdripper_t)
+term_list_ptys(cdripper_t)
+
+storage_raw_read_removable_device(cdripper_t)
+storage_read_scsi_generic(cdripper_t)
+storage_write_scsi_generic(cdripper_t)
+
+miscfiles_read_localization(cdripper_t)
+
+sysnet_dns_name_resolve(cdripper_t)
+
+userdom_use_user_terminals(cdripper_t)
+
+tunable_policy(`cdripper_rw_content && use_nfs_home_dirs',`
+	fs_list_auto_mountpoints(cdripper_t)
+	files_list_home(cdripper_t)
+	fs_manage_nfs_files(cdripper_t)
+	fs_manage_nfs_symlinks(cdripper_t)
+',`
+	files_dontaudit_list_home(cdripper_t)
+	fs_dontaudit_list_auto_mountpoints(cdripper_t)
+	fs_dontaudit_manage_nfs_files(cdripper_t)
+	fs_dontaudit_list_nfs(cdripper_t)
+')
+
+tunable_policy(`cdripper_rw_content && use_samba_home_dirs',`
+	fs_list_auto_mountpoints(cdripper_t)
+	files_list_home(cdripper_t)
+	fs_manage_cifs_files(cdripper_t)
+	fs_manage_cifs_symlinks(cdripper_t)
+',`
+	files_dontaudit_list_home(cdripper_t)
+	fs_dontaudit_list_auto_mountpoints(cdripper_t)
+	fs_dontaudit_manage_cifs_files(cdripper_t)
+	fs_dontaudit_list_cifs(cdripper_t)
+')
+
+tunable_policy(`cdripper_rw_content',`
+	userdom_list_user_tmp(cdripper_t)
+	userdom_manage_user_tmp_files(cdripper_t)
+	userdom_manage_user_tmp_symlinks(cdripper_t)
+	userdom_manage_user_home_content_files(cdripper_t)
+	userdom_manage_user_home_content_symlinks(cdripper_t)
+
+	ifndef(`enable_mls',`
+		fs_search_removable(cdripper_t)
+		fs_read_removable_files(cdripper_t)
+		fs_read_removable_symlinks(cdripper_t)
+	')
+',`
+	files_dontaudit_list_tmp(cdripper_t)
+	files_dontaudit_list_home(cdripper_t)
+	fs_dontaudit_list_removable(cdripper_t)
+	fs_dontaudit_read_removable_files(cdripper_t)
+	userdom_dontaudit_list_user_tmp(cdripper_t)
+	userdom_dontaudit_manage_user_tmp_files(cdripper_t)
+	userdom_dontaudit_list_user_home_dirs(cdripper_t)
+	userdom_dontaudit_write_user_home_content_files(cdripper_t)
+')
+
+tunable_policy(`use_nfs_home_dirs',`
+	files_search_mnt(cdripper_t)
+	fs_manage_nfs_files(cdripper_t)
+	fs_manage_nfs_symlinks(cdripper_t)
+')
+
```

Comment:
Let me get this straight: I what scenario would you be able to use "cdripper" without setting `cdripper_rw_content`? Were will you be ripping to?

Reply:
Don't know ++ don't care.

This is the way the cdrecord module was designed from the beginning and the aim of this patch is to enable the CD ripper functionality, not change the current module design.

This is a nack. But feel free to submit a separate patch, if you care so much about this issue and have time to spend developing and testing it.

Reply:
I am not asking you to interfere with the existing policy, but you cannot compare apples and oranges

Reply:
This is a pointless discussion !

First thing, I am not comparing different things !

Also, as already explained, I don't know where it is going to rip, but most importantly, **I don't care** !

It could rip to anywhere the underlying policy allows the calling domain to write, that's it.

If you want an example based on the current policy, then if the calling domain is _user_ it could write to USB removable media if the _user_write_removable_ boolean is turned ON.

Reply:
I agree with @doverride . There needs to be a default location to store the output. The home dir is the most obvious location. `cdripper_rw_content` should be removed and writes to the home content should be unconditional.